tests/test_s3.py

```diff
@@ -1910,6 +1910,244 @@ def test_delete_bucket_notification(self):
+            'filters': [
+                {'Name': bname}
+            ],
+            'actions': [{
+                'type': 'set-bucket-encryption'
+             }]
+            }, session_factory=session_factory)
+
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        # time.sleep(5)
+
+        response = client.get_bucket_encryption(Bucket=bname)
+        rules = response['ServerSideEncryptionConfiguration']['Rules'][0]['ApplyServerSideEncryptionByDefault']
+        self.assertEqual(rules['SSEAlgorithm'], 'AES256')
+
+        client.delete_bucket_encryption(Bucket=bname)
+        # time.sleep(5)
+
+        with self.assertRaises(Exception) as context:
+            client.get_bucket_encryption(Bucket=bname)
+
+        p = self.load_policy({
+            'name': 's3-enable-bucket-encryption',
+            'resource': 's3',
+            'filters': [
+                {'Name': bname}
+            ],
+            'actions': [{
+                'type': 'set-bucket-encryption',
+                'enabled': True,
+                'crypto': 'AES256'
+             }]
+            }, session_factory=session_factory)
+
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        # time.sleep(5)
+
+        response = client.get_bucket_encryption(Bucket=bname)
+        rules = response['ServerSideEncryptionConfiguration']['Rules'][0]['ApplyServerSideEncryptionByDefault']
+        self.assertEqual(rules['SSEAlgorithm'], 'AES256')
+
+    def test_delete_bucket_encryption(self):
+        self.patch(s3.S3, 'executor_factory', MainThreadExecutor)
+        self.patch(s3, 'S3_AUGMENT_TABLE', [])
+        session_factory = self.replay_flight_data('test_s3_delete_bucket_encryption')
+        session = session_factory()
+        client = session.client('s3')
+        bname = 'custodian-delete-bucket-encryption-aes256'
+
+        client.create_bucket(Bucket=bname)
+        self.addCleanup(destroyBucket, client, bname)
+
+        with self.assertRaises(Exception) as context:
+            response = client.get_bucket_encryption(Bucket=bname)
+
+        client.put_bucket_encryption(
+                Bucket=bname,
+                ServerSideEncryptionConfiguration={
+                    'Rules': [
+                        {
+                            'ApplyServerSideEncryptionByDefault': {
+                                'SSEAlgorithm': 'AES256'
+                                }
+                        }
+                    ]
+                })
+
+        p = self.load_policy({
+            'name': 's3-delete-bucket-encryption',
+            'resource': 's3',
+            'filters': [
+                {'Name': bname}
+            ],
+            'actions': [{
+                'type': 'set-bucket-encryption',
+                'enabled': False
+             }]
+            }, session_factory=session_factory)
+
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        # time.sleep(5)
```

Comment:
if sleeps are needed please put them behind if self.recording: conditional instead of commenting them out.